tests/test_vocabulary_handler.py

```diff
@@ -0,0 +1,275 @@
+
+    # Check that the xml doc has all the expected elements, values, and attributes.
+    assert root.tag == 'authority'
+    assert root.get('creator') == vocab.maintainer
+    assert root.get('created') == str(vocab.created).replace(' ', ', ')
+    assert root.get('modifier') == vocab.maintainer
+    assert root.get('modified') == str(vocab.modified).replace(' ', ', ')
+    assert root.enum.get('value') == term.name
+    assert root.enum.get('order') == '1'
+    assert root.enum.string.get('{http://www.w3.org/XML/1998/namespace}lang') == 'en'
+    assert root.enum.string == term.label
+    assert root.enum.linkback == prop.label
+    assert vocab_handler.vocab_mimetype == 'text/xml'
+    assert '<?xml version="1.0" encoding="UTF-8"?>' in vocab_handler.vocab_file
+
+
+def test_create_tkl_order_by_name():
+    vocab = factories.VocabularyFactory(order='name')
+    factories.TermFactory.create_batch(4, vocab_list=vocab)
+    vocab_handler = VocabularyHandler().tkl_response(vocab)
+
+    root = objectify.fromstring(vocab_handler.vocab_file)
+
+    sorted_terms = vocab.term_set.order_by('name')
+    for actual, expected in zip(root.enum, sorted_terms):
+        assert actual.get('value') == expected.name
+
+
+def test_create_tkl_order_by_label():
+    vocab = factories.VocabularyFactory(order='label')
+    factories.TermFactory.create_batch(4, vocab_list=vocab)
+    vocab_handler = VocabularyHandler().tkl_response(vocab)
+
+    root = objectify.fromstring(vocab_handler.vocab_file)
+
+    sorted_terms = vocab.term_set.order_by('label')
+    for actual, expected in zip(root.enum, sorted_terms):
+        assert actual.get('value') == expected.name
+
+
+def test_create_tkl_order_by_order():
+    vocab = factories.VocabularyFactory(order='order')
+    factories.OrderedTermFactory.create_batch(4, vocab_list=vocab)
+    vocab_handler = VocabularyHandler().tkl_response(vocab)
+
+    root = objectify.fromstring(vocab_handler.vocab_file)
+
+    sorted_terms = vocab.term_set.order_by('order', 'name')
+    for actual, expected in zip(root.enum, sorted_terms):
+        assert actual.get('value') == expected.name
+
+
+def test_create_vocab_dict():
+    # Create a vocab, term, and property that should be in the vocab_dict.
+    prop = factories.PropertyFactory()
+    vocab = prop.term_key.vocab_list
+
+    vocab_handler = VocabularyHandler().py_response(vocab)
+    vocab_dict = vocab_handler.create_vocab_dict('py')
+
+    assert vocab_dict['name'] == vocab.name
+    assert vocab_dict['label'] == vocab.label
+    assert vocab_dict['order'] == vocab.order
+    assert vocab_dict['maintainerEmail'] == vocab.maintainerEmail
+    assert vocab_dict['definition'] == vocab.definition
+    assert vocab_dict['created'] == vocab.created
+    assert vocab_dict['modified'] == vocab.modified
+    assert 'terms' in vocab_dict.keys()
+
+
+def test_create_vocab_dict_term_sub_dict():
+    # Test that the embedded dictionary contains the right information.
```

Comment:
Should this comment and some of the others in this file that are about the overall test function be docstrings?